compose: Avoid highlighting recipient elements when composing.

karlstolley opened the pull request:
This draft experimental PR matches recipient buttons and inputs to the compose box's background color except when a user is a) hovering over the area, after a brief delay (we want to avoid hover effects on incidental pointer traversal through the recipient row), or b) the user has focused the topic area, suggesting that they might be working with the topic in some way, whether editing, clearing, or simply just activating the topic box. Opening the channel picker and choosing a different channel likewise cancels the muted state.

As of 2025-06-17, the below issues have been addressed, and modified privacy-icon colors are behaving as expected--without any kind of intercession by JavaScript for hover or theme-switching states:

![alternate-privacy-icon-colors-compose](https://github.com/user-attachments/assets/ca6e5120-cb81-4539-a092-d90b5a150417)

As of 2025-06-16, the PR has a delicate commit for displaying darker privacy-icon colors in the muted recipient row. There are currently a couple important missing features, pending a decision that we want to move in this direction (I might also suggest that we calculate these modified colors up front when rendering decorate channels, so that there isn't a need for all the duplicative/derivative logic now present in the `maybe_mute_recipient_row()` function--instead we could just reach into some data attributes):

- [x] The darkened version of the icon remains even when hovering or activating the channel picker
- [x] Shifting from light mode to dark mode and vice-versa while the muted compose box is open does not update the modified icon color

As of 2025-06-09, the PR implements the [design adjustments](https://chat.zulip.org/#narrow/channel/101-design/topic/tighter.20compose.20box.20recipient.20row/near/2189209) as recommended by @terpimost:
* Transition delays and lengths have been shortened to feel snappier
* Hovering the new-topic button highlights the whole topic box

As of 2025-06-05, the PR brings DMs into participation with the muted-recipient-row logic.

As of 2025-05-23, this PR includes the commits from #34572, an experimental PR that has had some reasonable burn-in time on CZO.

Discussions:
* [#design > tighter compose box recipient row @ 💬](https://chat.zulip.org/#narrow/channel/101-design/topic/tighter.20compose.20box.20recipient.20row/near/2182073)
* [#design > making topic harder to edit? @ 💬](https://chat.zulip.org/#narrow/channel/101-design/topic/making.20topic.20harder.20to.20edit.3F/near/2157770)


_The Before & After shots here showcase adjustments to the icon and channel alignment within the channel picker._

| Before | After |
| --- | --- |
| ![short-channel-active-before](https://github.com/user-attachments/assets/3ce08fe3-6bd1-45a6-8476-c76405362d97) | ![short-channel-active-after](https://github.com/user-attachments/assets/99a34749-a633-4fb0-ac54-32c8431c328e) |
| ![short-channel-muted-before](https://github.com/user-attachments/assets/410e48bc-6315-40cf-b389-b1ada1d37453) | ![short-channel-muted-after](https://github.com/user-attachments/assets/618829c5-f5ac-4d47-9867-21a9bc9aedef) |
| ![long-channel-active-before](https://github.com/user-attachments/assets/a010b552-7852-47a1-8c91-02e8eb07076c) | ![long-channel-active-after](https://github.com/user-attachments/assets/70440a74-bf51-423c-a724-4fccda2bcda2) |
| ![long-channel-muted-before](https://github.com/user-attachments/assets/1bfc696c-c6ca-43ef-bb73-9b182e9d97cd) | ![long-channel-muted-after](https://github.com/user-attachments/assets/9865feb5-732c-40fa-987a-a43bcdd0e3fc) |

<details>
<summary>Self-review checklist</summary>

<!-- Prior to submitting a PR, follow our step-by-step guide to review your own code:
https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code -->

<!-- Once you create the PR, check off all the steps below that you have completed.
If any of these steps are not relevant or you have not completed, leave them unchecked.-->

- [x] [Self-reviewed](https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code) the changes for clarity and maintainability
      (variable names, code reuse, readability, etc.).

Communicate decisions, questions, and potential concerns.

- [x] Explains differences from previous plans (e.g., issue description).
- [x] Highlights technical choices and bugs encountered.
- [x] Calls out remaining decisions and concerns.
- [x] Automated tests verify logic where appropriate.

Individual commits are ready for review (see [commit discipline](https://zulip.readthedocs.io/en/latest/contributing/commit-discipline.html)).

- [x] Each commit is a coherent idea.
- [x] Commit message(s) explain reasoning and motivation for changes.

Completed manual review and testing of the following:

- [x] Visual appearance of the changes.
- [x] Responsiveness and internationalization.
- [x] Strings and tooltips.
- [x] End-to-end functionality of buttons, interactions and flows.
- [x] Corner cases, error conditions, and easily imagined bugs.
</details>

alya commented:
Notes from playing with this:

1. I would expect the channel picker to remain unfaded while you've got the menu open, regardless of mouse position.
![Screenshot 2025-04-15 at 15 10 45](https://github.com/user-attachments/assets/27902e3f-85de-4b73-a30e-f7e1950ac212)
2. I don't like the occasional effect where when you open the compose box, the recipient first appears unfaded, and then fades. This happened for me when I first opened the compose box when the app loaded, and a few other times, but I'm not sure what the pattern is.
3. Channel markers can be hard to see. In general, I kinda want the destination font to be less faded in light theme, at least, to make sure it doesn't get parsed as disabled.
 
![Screenshot 2025-04-15 at 15 14 53](https://github.com/user-attachments/assets/420c8324-4aa3-4fd7-9875-62ee03b53f07)
4. It felt like it would be more natural for the new topic button to operate more independently from the rest of the topic field. So, hovering over it would just activate the button, not the topic text input.
5. I think we want to hide the channel dropdown down arrow when things are faded. I dunno if the extra space will feel acceptable, or if we'll want the channel field to expand when clicked to accommodate it.
6. When things are faded, it feels like `>` is aligned too high with respect to the channel/topic names.

alya commented:
Ooops:

![Screenshot 2025-05-23 at 13 48 44@2x](https://github.com/user-attachments/assets/f3f2c229-ebc6-436a-9d3c-cf04737bf018)

karlstolley commented:
Corrected the palimpsest that was the DM picker button, and also implemented a parallel fix for the "Select channel" label when opting to start a new conversation.

alya commented:
I don't think the DMs logic is quite right. We shouldn't be fading when focus is in the addresee box, which seems to happen when you select a recipient form the typeahead with the keyboard.

![Screenshot 2025-06-05 at 14 55 11](https://github.com/user-attachments/assets/aa941590-d1e3-43a8-84f6-363a8f996f99)

alya commented:
Also, the fading doesn't seem to happen if I _remove_ compose box recipients until they match the current conversation view.

karlstolley commented:
The first one is definitely a bug, but the latter is not, I don't think: IIRC, we made a decision in #design that once a user starts interacting with the recipient row, we keep it highlighted--even if they return to the state from when then box first opens.

alya commented:
Oh, hm, OK.

alya commented:
The latest version feels fine to me in manual testing!

timabbott commented on web/src/compose_actions.ts:
```diff
@@ -91,6 +91,10 @@ function hide_box(): void {
     $("#compose_controls").show();
+    // Assume a muted recipient row for the next time
+    // the compose box is reopened
+    $("#compose-recipient").addClass("muted-recipient-row");
```
I think "muted" is the wrong term to use here; just because a "muted recipient" is a thing. 

Maybe "recipient-row-attention-low" would work?

timabbott replied:
I guess then `maybe_mute_recipient_row` would be `update_recipient_row_attention`?

I think this can be a follow-up rename from my standpoint.

timabbott commented:
Didn't see anything else when finishing this review, merged, thanks for all the work on this @karlstolley! Definitely quite a few rabbit holes were involved in getting there :)
